src/NinjaTools.FluentMockServer.API/Proxy/Visitors/IVisitor.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using NinjaTools.FluentMockServer.API.Proxy.Visitors.Collections;
+using NinjaTools.FluentMockServer.API.Types;
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+
+namespace NinjaTools.FluentMockServer.API.Proxy.Visitors
+{
+    public class RequestBodyMatcher :  IDontRenderWhenEmpty
+    {
+        public string? Content { get; set; }
+        public RequestBodyKind?  Kind { get; set; }
+        public bool? MatchExact { get; set; }
+
+        /// <inheritdoc />
+        public bool IsEnabled() => !string.IsNullOrWhiteSpace(Content) || Kind.HasValue || MatchExact.HasValue;
+    }
+
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum RequestBodyKind
+    {
+        Text,
+        Base64
+    }
+
+
+ //   [JsonObject(MemberSerialization.OptOut)]
+    public class RequestMatcher
+    {
+
+        [YamlMember(SerializeAs = typeof(Dictionary<string, object>))]
+        public RequestBodyMatcher BodyMatcher { get; set; } = new RequestBodyMatcher();
+
+
+        [YamlMember(SerializeAs = typeof(string), ScalarStyle = ScalarStyle.Any)]
+        public PathCollection Path { get; set; } = new PathCollection(string.Empty);
+
+
+        [YamlMember(SerializeAs = typeof(string), ScalarStyle = ScalarStyle.Any)]
+        public HttpMethodWrapper Method { get; set; } = new HttpMethodWrapper();
+
+
+        [YamlMember(SerializeAs = typeof(Dictionary<string, string[]>))]
+        public HeaderCollection Headers { get; set; } = new HeaderCollection();
+
+
+        [YamlMember(SerializeAs = typeof(Dictionary<string, string>))]
+        public CookieCollection Cookies { get; set; } = new CookieCollection();
+
+
+        [YamlMember(SerializeAs = typeof(Dictionary<string, string[]>))]
+        public QueryCollectionWrapper Query { get; set; } = new QueryCollectionWrapper();
+    }
+
+    public interface IVisitor
+    { }
+
+    public interface IVisitor<TVisitable>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add the 'in' keyword to parameter 'TVisitable' to make it 'contravariant'.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4802644)